proc.c

```diff
@@ -2739,6 +2739,18 @@ rb_obj_method_arity(VALUE obj, ID id)
     return rb_mod_method_arity(CLASS_OF(obj), id);
 }
 
+VALUE
+rb_callable_receiver(VALUE callable) {
+    if (rb_obj_is_proc(callable)) {
+        VALUE binding = rb_funcall(callable, rb_intern("binding"), 0);
```

Comment:
do not depend Proc#binding because it can be limited in a future.

Reply:
Hmm not sure why I didn't call the methods directly since it's in the same compilation unit. I'll try that.